python/paddle/fleet/meta_optimizers/amp_optimizer.py

```diff
@@ -0,0 +1,63 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+import paddle.fluid.contrib.mixed_precision as mixed_precision
+from .meta_optimizer_base import MetaOptimizerBase
+
+__all__ = ["AMPOptimizer"]
+
+
+class AMPOptimizer(MetaOptimizerBase):
+    def __init__(self, optimizer):
+        super(AMPOptimizer, self).__init__(optimizer)
+        self.inner_opt = optimizer
+        self.amp_opt = None
+        # we do not allow meta optimizer to be inner optimizer currently
+        self.meta_optimizers_white_list = []
+
+    def _set_basic_info(self, loss, role_maker, user_defined_optimizer,
+                        user_defined_strategy):
+        super(AMPOptimizer, self)._set_basic_info(
+            loss, role_maker, user_defined_optimizer, user_defined_strategy)
+
+    def _can_apply(self):
+        if self.user_defined_strategy.amp:
+            return True
+        return False
+
+    def _disable_strategy(self, dist_strategy):
+        dist_strategy.amp = False
+
+    def minimize_impl(self,
+                      loss,
+                      startup_program=None,
+                      parameter_list=None,
+                      no_grad_set=None):
+        if self.amp_opt is None:
```

Comment:
In what cases will the "self.amp_out" be not None?

Reply:
I think no case, added this for safety

Reply:
https://github.com/PaddlePaddle/Paddle/blob/5e07db15e66710a63e46cc874756db44a9f89f78/python/paddle/fluid/contrib/mixed_precision/decorator.py#L62-L64
Because OptimizerWithMixedPrecision.\_\_init\_\_ will add some var and op, may pollute Program, so init `amp_opt` in `minimize_impl`. These var and op should be removed from \_\_init\_\_ to compatible with fleet. But this changes the code implementation, so I'm going to change it in the next PR.